docs/proposals/20191030-machine-health-checking.md

```diff
@@ -108,6 +109,11 @@ As an operator of a Management Cluster, I want my machines to be self-healing an
 
 ### Implementation Details/Notes/Constraints
 
+#### Machine annotation:
+```go
+const MachineUnhealthyAnnotation = "machine.cluster.x-k8s.io/unhealthy"
```

Comment:
I'm wondering if this should be have a time as value?

Reply:
We need to expand this section to give some guarantees, for example if this is a timestamp, do we keep updating the annotation at every reconciliation?

Reply:
As an aside, to support different external remediation (outside of the owning controller), such as the request from #2846, would it make sense to allow for overriding the annotation key here?

Reply:
Having different timestamps, while interesting, also means unfortunately that the Machine and everything that watches them is going to be updated every time the check is performed :/

>to support different external remediation (outside of the owning controller), such as the request from #2846, would it make sense to allow for overriding the annotation key here?

I'd expect that at some point (proposal tbd) that we'll allow different strategies for remediation.

Maybe to start, we should have a simple annotation with no timestamp on it?

Reply:
Is it possible to have also an annotation that signals when the machine is not healthy but the timeout is not yet reached (machine under probation)

Reply:
I'd say that until we have a use case for it being a timestamp, let's not add that extra complexity

Reply:
Also, you can't trust it